og_ui/src/BundleFormAlter.php

```diff
@@ -0,0 +1,171 @@
+    $this->addGroupContent($form, $form_state);
+  }
+
+  public function ajaxCallback(array $form, array &$form_state) {
+    return $form['og']['target_bundles'];
+  }
+
+  protected function prepare(array &$form, $form_state) {
+    // Example: article.
+    $this->bundle = $this->entity->id();
+    // Example: Article.
+    $this->bundleLabel = Unicode::lcfirst($this->entity->label());
+    $this->definition = \Drupal::entityTypeManager()->getDefinition($this->entity->getEntityTypeId());
+    // Example: node.
+    $this->entityTypeId = $this->definition->getBundleOf();
+
+    $form['og'] = array(
+      '#type' => 'details',
+      '#title' => t('Organic groups'),
+      '#collapsible' => TRUE,
+      '#group' => 'additional_settings',
+      '#description' => t('Each "%bundle" may serve as a group, may belong to a group, or may not participate in OG at all.', [
+        '%bundle' => $this->bundleLabel,
+      ]),
+    );
+
+  }
+
+  protected function addGroupType(array &$form, $form_state) {
+    $form['og']['og_is_group'] = array(
+      '#type' => 'checkbox',
+      '#title' => t('Group'),
+      '#default_value' => Og::isGroup($this->entityTypeId, $this->bundle),
+      '#description' => t('Every "%bundle" is a group which can contain entities and can have members.', [
+        '%bundle' => Unicode::lcfirst($this->bundleLabel),
+      ]),
+    );
+  }
+
+  protected function addGroupContent(array &$form, $form_state) {
+    $is_group_content = Og::isGroupContent($this->entityTypeId, $this->bundle);
+
+    $target_type_default = FALSE;
+    $handler_settings = [];
+    if ($field = FieldConfig::loadByName($this->entityTypeId, $this->bundle, OG_AUDIENCE_FIELD)) {
+      $handler_settings = $field->getSetting('handler_settings');
+      if (isset($handler_settings['target_type'])) {
+        $target_type_default = $handler_settings['target_type'];
+      }
+    }
+
+    $target_types = [];
+    $bundle_options = [];
+    $all_group_bundles = Og::groupManager()->getAllGroupBundles();
+    foreach ($all_group_bundles as $group_entity_type => $bundles) {
+      if (!$target_type_default) {
+        $target_type_default = $group_entity_type;
+      }
+      $target_types[$group_entity_type] = \Drupal::entityTypeManager()
+        ->getDefinition($group_entity_type)
+        ->getLabel();
+    }
+
+    if ($all_group_bundles) {
+      $bundle_info = \Drupal::service('entity_type.bundle.info')
+        ->getBundleInfo($target_type_default);
+      foreach ($all_group_bundles[$target_type_default] as $bundle_name) {
+        $bundle_options[$bundle_name] = $bundle_info[$bundle_name]['label'];
+      }
+      $description = '';
+    }
+    else {
+      $description = t('There are no group bundles defined.');
+    }
+
+    $form['og']['og_group_content_bundle'] = array(
+      '#type' => 'checkbox',
+      '#title' => t('Group content'),
+      '#default_value' => $is_group_content,
+      '#description' => $description,
+    );
```

Comment:
This field is managed here not on the field UI so there's no more `#disabled` here, unchecking the checkbox will delete the field here. If there are no `$target_types` we disable below, where it logically below.